### PR TITLE
Obmin 274 product card

### PR DESCRIPTION
**Pull request by Kyslytsia**
Created Product card, Inbox message

### CODE REVIEW

**Review by AkintsevaElizaveta — APPROVED** [review]

**Review by Voloshkina — APPROVED** [review]

**Review by Viktoria-Serkiz — APPROVED** [review]

File: src/components/ellipsis-text/EllipsisText.stories.tsx [hunk 1]
```diff
@@ -31,5 +31,5 @@ WithAnotherTooltipDirection.args = {
   children: text,
   delayShow: 1000,
   delayHide: 1000,
-  place: 'right-start',
+  place: 'right',
```
Comment (cloudmenson) [inline src/components/ellipsis-text/EllipsisText.stories.tsx:34]: don't forget to sort by length

File: src/components/ellipsis-text/styles.ts [hunk 2]
```diff
@@ -12,15 +12,11 @@ export const ReactTooltip = styled(Tooltip).attrs({
...
-      box-shadow: ${theme.colors.ellipsisShadow};
-    `}
-  }
-
-  &.custom-arrow {
-    background: transparent;
+      ${({ theme }) => css`
+        background-color: ${theme.colors.white};
+        color: ${theme.colors.blackColorText};
+        box-shadow: ${theme.colors.ellipsisShadow};
```
Comment (cloudmenson) [inline src/components/ellipsis-text/styles.ts:18]: here you can also sort by length for better reading

File: src/components/product-card/ProductCard.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+export const ProductCardWithItem = Template.bind({});
+ProductCardWithItem.args = {
+  city: 'NEW YORK',
+  text: 'hoody',
+  picture:
+    'https://gfx.garnamama.com/pub/products/235/196235/1600x1600/2EI~124024-7.jpg',
+  isFavorite: true,
+  avatar: image,
+  inboxMessage: 9,
+  buttonText: 'button',
```
Comment (cloudmenson) [inline src/components/product-card/ProductCard.stories.tsx:26]: Sort by length pls

File: src/components/product-card/styles.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  text-align: center;
+`;
+
+export const CitySpan = styled.span`
+  position: relative;
+  max-width: 120px;
+  margin-left: 8px;
+  font-size: 14px;
+  z-index: 1;
+  color: ${({ theme: { colors } }) => colors.btnBlueHover};
```
Comment (cloudmenson) [inline src/components/product-card/styles.ts:101]: do the same as on line 77

File: src/config/theme/colors.ts [hunk 5]
```diff
@@ -7,9 +7,11 @@ const buttonBlueHover = '#00bfff';
 const buttonBlueActive = '#0094ff';
 
 //Green
+const apple = '#37B82E';
 const buttonGreen = '#43b15d';
-const buttonGreenHover = '#3c9f53';
 const buttonGreenActive = '#008a20';
+const buttonGreenHover = '#3c9f53';
+const forestGreen = '#32a72a';
```
Comment (cloudmenson) [inline src/config/theme/colors.ts:14]: Sort by length

File: src/config/theme/colors.ts [hunk 6]
```diff
@@ -22,21 +24,60 @@ const rightColorText = '#333333';
 const colorBorderFocus = '#707070';
 const colorTextDisabled = '#A3A3A3';
 
+//General yellow
+const corn = '#E0C600';
+const schoolBusYellow = '#f5d800';
+const turbo = '#FEE200';
```
Comment (cloudmenson) [inline src/config/theme/colors.ts:30]: Same

File: src/config/theme/colors.ts [hunk 7]
```diff
@@ -22,21 +24,60 @@ const rightColorText = '#333333';
...
+
+//General grey
+const dustyGray = '#969696';
+const silver = '#c4c4c4';
+const silverChalice = '#A0A0A0';
```
Comment (cloudmenson) [inline src/config/theme/colors.ts:35]: Same

File: src/config/theme/colors.ts [hunk 8]
```diff
@@ -22,21 +24,60 @@ const rightColorText = '#333333';
...
+  error: colorError,
   focus: colorBorderFocus,
-  textBlack: blackColorText,
-  textRight: rightColorText,
   searchBorder: activeColor,
+  searchBtn: buttonBlue,
+  textBlack: blackColorText,
   textDisabled: colorTextDisabled,
+  textGrey: colorGrey,
+  textRight: rightColorText,
```
Comment (cloudmenson) [inline src/config/theme/colors.ts:46]: Sort by length

File: src/config/theme/colors.ts [hunk 9]
```diff
@@ -22,21 +24,60 @@ const rightColorText = '#333333';
...
+});
+
+const inboxMessage = Object.freeze({
+  inboxMessageActive: forestGreen,
+  inboxMessageHover: apple,
```
Comment (cloudmenson) [inline src/config/theme/colors.ts:51]: Sort by length

File: src/config/theme/colors.ts [hunk 10]
```diff
@@ -22,21 +24,60 @@ const rightColorText = '#333333';
...
+});
+
+const itemCard = Object.freeze({
+  avatarFalseDefault: silverChalice,
+  avatarFalseHover: dustyGray,
+  avatarTrueActive: corn,
+  avatarTrueDefault: turbo,
+  avatarTrueHover: schoolBusYellow,
```
Comment (cloudmenson) [inline src/config/theme/colors.ts:59]: Sort by length

File: src/config/theme/colors.ts [hunk 11]
```diff
@@ -22,21 +24,60 @@ const rightColorText = '#333333';
...
 
 const ellipsisShadow =
   '10px 10px 20px rgba(0, 0, 0, 0.3), 0px 31px 32px -24px rgba(0, 0, 0, 1)';
 
+const button = {
+  blue: {
+    bg: buttonBlue,
+    hover: buttonBlueHover,
+    active: buttonBlueActive,
+    disable: buttonDisabled,
```
Comment (cloudmenson) [inline src/config/theme/colors.ts:71]: The same

File: src/config/theme/colors.ts [hunk 12]
```diff
@@ -22,21 +24,60 @@ const rightColorText = '#333333';
...
+  },
+  green: {
+    bg: buttonGreen,
+    hover: buttonGreenHover,
+    active: buttonGreenActive,
+    disable: buttonDisabled,
```
Comment (cloudmenson) [inline src/config/theme/colors.ts:77]: Same

File: src/components/ellipsis-text/styles.ts (hunk 2 above)
Comment (Kyslytsia) [reply src/components/ellipsis-text/styles.ts:18]: done

File: src/components/ellipsis-text/EllipsisText.stories.tsx (hunk 1 above)
Comment (Kyslytsia) [reply src/components/ellipsis-text/EllipsisText.stories.tsx:34]: done

File: src/components/product-card/ProductCard.stories.tsx (hunk 3 above)
Comment (Kyslytsia) [reply src/components/product-card/ProductCard.stories.tsx:26]: done

File: src/components/product-card/styles.ts (hunk 4 above)
Comment (Kyslytsia) [reply src/components/product-card/styles.ts:101]: done

**Review by cloudmenson — APPROVED** [review]

**Review by ASGladchenko — APPROVED** [review]

**Review by Viktoria-Serkiz — APPROVED** [review]

File: src/components/product-card/index.tsx [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+const ProductCard = ({
+  city,
+  text,
+  onClick,
+  picture,
+  buttonText,
+  isFavorite,
+  margin = 0,
+  avatar = '',
+  inboxMessage,
```
Comment (ASGladchenko) [inline src/components/product-card/index.tsx:18]: In the style of the Card component, you have:
margin:  ${margin}  ?  ${margin}px  ||  0;
maybe you don't need to assign 0 here

File: src/components/product-card/types.ts [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+  picture: string;
+  buttonText: string;
+  onClick: () => void;
+  isFavorite?: boolean;
+  inboxMessage?: number;
+}
+
+export interface InboxMessageProps {
+  inboxMessage: number;
+}
```
Comment (ASGladchenko) [inline src/components/product-card/types.ts:15]: This interface you use in src/components/inbox-message/index.tsx
for this it was created in
src/components/inbox-message/types.ts
maybe it's not needed here?

Comment (Kyslytsia) [reply src/components/product-card/types.ts]: That is how it should be

**Review by Voloshkina — APPROVED** [review]

**Pull request merged**